allennlp/notebooks/vocabulary.ipynb

```diff
@@ -0,0 +1,293 @@
+    "\n",
+    "\n",
+    "It's also easy to interact with the Vocabulary to retrieve specific word ids or tokens for a given id.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Index 2 has token:  Paul\n",
+      "Word 'Allen' has index:  3\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Index 2 has token: \", vocab.get_token_from_index(2, namespace=\"tokens\"))\n",
+    "print(\"Word 'Allen' has index: \", vocab.get_token_index(\"Allen\", namespace=\"tokens\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Also note that `namespaces` will deal with OOV tokens differently depending on whether they contain an OOV Token. See the difference between asking for an index for \"pernacious\" in the two different `namespaces` we've created in our toy `Vocabulary`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The index of 'pernacious' in the 'tokens' namespace should be 1 (The @@UNKOWN@@ token):  1\n",
+      "As 'tags' doesn't have an unknown token, fetching non-existent tags will throw a KeyError.\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"The index of 'pernacious' in the 'tokens' namespace should be 1 (The @@UNKOWN@@ token): \", vocab.get_token_index(\"pernacious\", namespace=\"tokens\"))\n",
+    "\n",
+    "try:\n",
+    "    vocab.get_token_index(\"pernacious\", namespace=\"tags\")\n",
+    "except KeyError:\n",
+    "    print(\"As 'tags' doesn't have an unknown token, fetching non-existent tags will throw a KeyError.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true
+   },
+   "source": [
+    "\n",
+    "Above, we demonstrated the basic functionality of the namespaces in the Vocabulary. So far so good - probably not much different to other `Vocabulary` type classes for NLP that you've seen before. However, we'd ideally like to \n",
+    "generate a full `Vocabulary` without having to individually add all the different words. Below, we'll generate a `Dataset` consisting of a single `Instance` and use it to automatically generate a `Vocabulary`. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from allennlp.data.fields import TextField, TagField\n",
+    "from allennlp.data import Dataset, Instance\n",
+    "from allennlp.data.token_indexers import SingleIdTokenIndexer\n",
+    "sentence = TextField(tokens=[\"Paul\", \"Allen\", \"is\", \"a\", \"great\", \"guy\", \".\"],\n",
```

Comment:
...  I mean, I'm not saying Paul Allen's not a great guy - he is funding us - but this just gives an odd tone to the tutorial, I think.